repair_refurbish_repair_stock_move/models/repair_order.py

```diff
@@ -13,6 +13,37 @@ def create_refurbished_stock_move(self):
         move.quantity_done = self.product_qty
         self.refurbish_move_id = move.id
 
+    def _create_and_confirm_stock_moves(self):
+        res = super()._create_and_confirm_stock_moves()
+        # if the repair order is started the action assign needs to be called
+        for rec in self.filtered(lambda l: l.state == "under_repair"):
+            moves = rec.mapped("stock_move_ids")
+            moves.filtered(lambda l: l.state == "confirmed")._action_assign()
+        return res
+
+    def write(self, values):
+        res = super().write(values)
+        if "to_refurbish" in values.keys():
+            if self.mapped("stock_move_ids") and self.state not in (
+                "draft",
+                "done",
+                "cancel",
+            ):
+                # recreate stock moves
+                self.mapped("stock_move_ids")._action_cancel()
+                self._create_and_confirm_stock_moves()
```

Comment:
we should evaluate this in a loop for rec in self to make sure the state of each record being written is evaluated

Reply:
you are right! thanks! it is done now.